### PR TITLE
Fix remaining check warnings and undeclared dependencies from PEcAn.utils 

### PR DESCRIPTION
**Pull request by moki1202**

### CODE REVIEW

**Comment by moki1202** [comment]
I forgot to create this PR as a draft, apologies.

File: base/utils/DESCRIPTION
Comment (infotroph) [inline base/utils/DESCRIPTION:8]:
```suggestion
           person("Mike", "Dietze", role = "aut"),
           person("Rob","Kooper", role = c("aut", "cre"), 
```

**Review by infotroph — COMMENTED** [review]
Incomplete review, will try to finish tomorrow, but I am PUMPED to see these deprecated functions going away! Let's make sure we record those in the changelog in this PR and close their tracking issues (I'll find and link in comments) once this is merged

File: base/utils/man/n.Rd
Comment (infotroph) [reply base/utils/man/n.Rd:4]: Found where this came from: https://github.com/PecanProject/pecan/pull/2830#discussion_r680258637

**Comment by infotroph** [comment]
Issues this PR will close: #2169,  #2065, #2031

Issues it won't definitely close but that we should review and think about how they connect: #2425, #1722, #1681, and probably others... this package has needed attention for a while!

File: base/utils/R/utils.R
Comment (moki1202) [reply base/utils/R/utils.R:208]: @infotroph yes I tried initializing n and statname as NULL arguments in the summarize.result function but kept getting the same note.

Comment (moki1202) [reply base/utils/R/utils.R:208]: I might have a solution for this.

Comment (moki1202) [reply base/utils/R/utils.R:208]: I am an idiot! n and statname never caused any "global variable" NOTE. I'll revert this.

File: base/utils/R/sensitivity.R
Comment (infotroph) [reply base/utils/R/sensitivity.R:39]: Let's revert now and handle in the next PR.
```suggestion
      sa.run.ids <- runs.samples$sa
```

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R:27]:
```suggestion
##' @param silent logical: suppress log messages about missing variables?
```

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R:696]:
```suggestion
##' @param isError function to use for checking whether to try again.
##'   Must take one argument that contains the result of evaluating `expr`
##'   and return TRUE if another retry is needed 
```

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R:702]:
```suggestion
##'   file_url <- paste0("https://thredds.daac.ornl.gov/", 
##'       "thredds/dodsC/ornldaac/1220", 
##'       "/mstmip_driver_global_hd_climate_lwdown_1999_v1.nc4")
##' dap <- retry.func(
```

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R:706]:
```suggestion
##'   ncdf4::nc_open(file_url)
```

**Review by infotroph — CHANGES_REQUESTED** [review]
Great improvements! This cleanup was a beast; thank you for taking it on.

There are a few changes here that I think still need more thought (mostly "no visible binding for global variable" patches that I think are caused by real undefined variable bugs -- setting NULL would just suppress the warning without resolving the problem). I've made suggestions that revert these changes and I propose we start a separate PR that handles these by themselves.

Still to do before merging this:
* Document these changes in NEWS.md, and *also* record the deleted and moved functions in the whole-PEcAn CHANGELOG.md to make sure they're prominently visible to everyone.
* Search the rest of PEcAn for uses of the deleted functions. I hope you won't find any, but if they exist update them to the appropriate replacements from other packages.
* Search the rest of PEcAn for uses of the relocated functions, and update to call them from their new packages.

File: base/utils/R/get.analysis.filenames.r
Comment (moki1202) [reply base/utils/R/get.analysis.filenames.r:4]: @infotroph removing @name <> and @title <> won't cause any troubles in the documentation files?

Comment (infotroph) [reply base/utils/R/get.analysis.filenames.r:4]: Nope, because Roxygen defaults to using the function name for @name and the first line of the comment block for @title, and both of those are identical here so it's cleaner to avoid duplicating.

Comment (moki1202) [reply base/utils/R/get.analysis.filenames.r:4]: yep got it!

Comment (infotroph) [reply base/utils/R/get.analysis.filenames.r]: (You can verify this by looking at the generated `man/ensemble.filename.Rd` before and after making the change -- the `\name{..}` and `\title{...}` lines shouldn't change.)

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R]: I this this is undoing an earlier fix -- this line should be retained

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R]: backticks should be around whole `options(...)`

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R]: don't delete

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R]: undoes previous fix

**Review by infotroph — APPROVED** [review]

File: base/utils/.Rbuildignore
Comment (infotroph) [inline base/utils/.Rbuildignore:3]:
```suggestion
^scripts

```

**Comment by infotroph** [comment]
@moki1202 are we OK to change this PR from draft to ready-for-review?

File: base/utils/R/utils.R
Comment (infotroph) [inline base/utils/R/utils.R:44]:
```suggestion

```

**Comment by moki1202** [comment]
@infotroph there are still 1-2 fixes remaining for utils but I think we can do that in another PR.

**Comment by infotroph** [comment]
@moki1202 I agree. Any remaining changes in this PR will be ones that are requested by reviewers other than me or that are needed to make the CI build pass, so I'll mark it "ready for review" to let other folks know they can start reading it.

File: base/utils/R/convert.input.R
Comment (mdietze) [inline base/utils/R/convert.input.R:59]: I'd recommend putting the args in the same order in the documentation as they are in the function, and definitely with `...` last. Also, seems like we should say what goes in the `dbparms` list (or at least crossreference someplace that does)

File: base/utils/R/get.analysis.filenames.r
Comment (mdietze) [inline base/utils/R/get.analysis.filenames.r:72]: maybe I'm misremembering, but weren't you moving the sensitivity functions to the uncertainty package?

File: base/utils/R/sensitivity.R
Comment (mdietze) [inline base/utils/R/sensitivity.R:27]: same question as above about where sensitivity functions should live

File: base/utils/R/get.analysis.filenames.r
Comment (infotroph) [reply base/utils/R/get.analysis.filenames.r:72]: We are but we missed this file. @moki1202 I think this will be resolved when you merge https://github.com/moki1202/pecan/pull/5 (assuming you agree with my fix!)

File: models/biocro/DESCRIPTION
Comment (infotroph) [inline models/biocro/DESCRIPTION:19]:
```suggestion
```

File: Makefile.depends
Comment (infotroph) [inline Makefile.depends:26]:
```suggestion
$(call depends,models/biocro): | .install/base/logger .install/base/remote .install/base/utils .install/base/settings .install/modules/data.atmosphere .install/modules/data.land .install/base/db
```

**Comment by infotroph** [comment]
Whew, all checks passing!

**Pull request merged**
